plutus-ledger-api/src/Plutus/ApiCommon.hs

```diff
@@ -74,18 +105,164 @@ builtinsIntroducedIn = Map.fromList [
           ChooseData, ConstrData, MapData, ListData, IData, BData, UnConstrData, UnMapData, UnListData, UnIData, UnBData, EqualsData,
           MkPairData, MkNilData, MkNilPairData
           ]),
-  (ProtocolVersion 6 0, Set.fromList [
+  ((PLC.Version () 2 0 0, ProtocolVersion 6 0), Set.fromList [
           SerialiseData, VerifyEcdsaSecp256k1Signature, VerifySchnorrSecp256k1Signature
           ])
   ]
 
 -- | Which builtin functions are available in the given 'ProtocolVersion'?
 --
 -- See Note [New builtins and protocol versions]
-builtinsAvailableIn :: ProtocolVersion -> Set.Set PLC.DefaultFun
-builtinsAvailableIn pv = fold $ Map.elems $ Map.takeWhileAntitone (\introducedInPv -> introducedInPv <= pv) builtinsIntroducedIn
+builtinsAvailableIn :: LanguageVersion -> ProtocolVersion -> Set.Set PLC.DefaultFun
+builtinsAvailableIn lv pv = fold $ Map.elems $ Map.takeWhileAntitone (\(introducedInLv, introducedInPv) -> introducedInLv <= lv && introducedInPv <= pv) builtinsIntroducedIn
```

Comment:
Okay, so you're making the assumption that the builtins in later languages are always a superset of those in previous languages. I think we shouldn't make that assumption, since one of the most useful things we can do in a new language version is _remove_ a builtin! That may mean we just need to duplicate `builtinsIntroducedIn` for the two language versions...

Reply:
It is the exact same assumption we make for the protocolversion, and something i pointed out last time as well. I think we should leave this for now.

Reply:
See: https://github.com/input-output-hk/plutus/pull/4416#discussion_r821484051

Reply:
Hah. You're quite right, I'm being inconsistent.